src/AiSystem.cpp

```diff
@@ -104,6 +75,112 @@ void AiSystem::handle_new_it(const Event& e) {
+            if (car == i) {
+                continue;
+            }
+
+            float distance = glm::distance(cars_[car].position_, cars_[i].position_);
+
+            if (distance < closest_car_distance) {
+                closest_car = i;
+                closest_car_distance = distance;
+            }
+        }
+
+        if (closest_car_distance < AVOID_DISTANCE) {
+            avoid_car(car, closest_car);
+        } else {
+            random_car(car);
+        }
+    } else {
+        auto pos1 = cars_[car].position_;
+        auto pos2 = cars_[whos_it_].position_;
+
+        cars_[car].cutoff_cooldown_--;
+
+        if (counter_ == 0) {
+            if (glm::distance(pos1, pos2 * .75f) > 2.5f && cars_[car].cutoff_cooldown_ > 0) {
+                cars_[car].state_ = AiState::Cutoff;
+                cars_[car].cutoff_cooldown_ = 300;
+            }
+        }
+
+        if (glm::distance(pos1, pos2 * .75f) < 2.5f && cars_[car].state_ == AiState::Cutoff) {
+            cars_[car].cutoff_cooldown_ = 300;
+            cars_[car].state_ = AiState::Pursue;
+        }
+
+
+
+        if (cars_[car].state_ == AiState::Cutoff) {
+            cutoff_car(car, whos_it_);
+        } else {
+            pursue_car(car, whos_it_);
+        }
+    }
+}
+
+void AiSystem::avoid_car(int car, int to_avoid) {
+    glm::vec3 diff = cars_[to_avoid].position_ - cars_[car].position_;
+    glm::vec3 target = glm::rotate(diff, glm::radians(90.f), glm::vec3(0, 1, 0));
+
+    path_to(car, target);
+}
+
+void AiSystem::pursue_car(int car, int to_pursue) {
+    float aheadness = glm::distance(cars_[to_pursue].position_, cars_[car].position_);
+    glm::vec3 target = cars_[to_pursue].position_ + cars_[to_pursue].rotation_ * glm::vec3(0, 0, 0.5f + aheadness);
+
+    path_to(car, target);
+}
+
+void AiSystem::random_car(int car) {
+    if (counter_ % 10 == 0) {
+        unsigned int random = rand() % 20;
+
+        if (random == 0) {
+            EventSystem::queue_event(
+                Event(
+                    EventType::KEYPRESS_EVENT,
+                    "player_id", car,
+                    "key", SDL_CONTROLLER_AXIS_LEFTX,
+                    "value", -32768
+                )
+            );
+        } else if (random == 1) {
+            EventSystem::queue_event(
+                Event(
+                    EventType::KEYPRESS_EVENT,
+                    "player_id", car,
+                    "key", SDL_CONTROLLER_AXIS_LEFTX,
+                    "value", 32768
+                )
+            );
+        } else {
+            EventSystem::queue_event(
+                Event(
+                    EventType::KEYPRESS_EVENT,
+                    "player_id", car,
+                    "key", SDL_CONTROLLER_AXIS_TRIGGERRIGHT,
+                    "value", 32768
+                )
+            );
+        }
+    }
+}
+
+void AiSystem::cutoff_car(int car, int to_catch) {
+    glm::vec3 target = cars_[to_catch].position_;
+    target *= 0.75f;
```

Comment:
What's the thought process for this? I don't get how it would work, but I'm gonna guess you're smarter than I am :)